flume-ng-node/src/main/java/org/apache/flume/node/AbstractConfigurationProvider.java

```diff
@@ -364,6 +358,57 @@ private void loadSources(AgentConfiguration agentConf,
     }
   }
 
+  private List<Channel> getSourceChannels(Map<String, ChannelComponent> channelComponentMap,
+                                          Source source, Collection<String> channelNames) {
+    List<Channel> sourceChannels = new ArrayList<Channel>();
+    for (String chName : channelNames) {
+      ChannelComponent channelComponent = channelComponentMap.get(chName);
+      if (channelComponent != null) {
+        try {
+          checkSourceChannelCompatibility(source, channelComponent.channel);
+          sourceChannels.add(channelComponent.channel);
+        } catch (InstantiationException e) {
```

Comment:
If there is a problem with one of the channels this would just silently remove that one. I think we should let the exception out and make the whole Source removed from the config.

Reply:
I agree.